### PR TITLE
[READY]Refactors reagent container types

### LINKED ISSUE #33518: XENOBIO SLIME EXTACTS ARE BROKEN
The following pull requests are currently test merged: 
#33460: 'Re-Support old hotkey mode behaviour' by ninjanomnom at commit d0e455 
#33470: 'Refactors reagent container types' by ACCount12 at commit 9de85e 
#33480: 'Deadmin tweaks: Admins without +AUTOLOGIN start deadmined. AUTOLOGIN defaults to on.' by MrStonedOne at commit 2ac9d8 
#33465: 'Replaces a bunch of loc assignments with forcemoves and moves to nullspace' by vuonojenmustaturska at commit b82ed5 

you cant extract chemicals from the slime extracts

### PR DESCRIPTION
**Pull request by ACCount12**
This PR standardizes the way most reagent containers check if it's possible to transfer from one container to another. It's now possible to control if the user is able to inject, draw, pour, refill, see the contents or the amount of contents of any given container.

* Fixes being able to extract reagents out of food with syringe.
* Breaks #33518. Fixes #33518.
* Reagent dispensers (water tank, etc) no longer rely on snowflake checks, it's now possible to make any container have dispenser-like behavior.
* New type flag: `AMOUNT_VISIBLE`. Used for objects that have only the general amount of reagents visible on examine. Fire extinguishers and reagent dispensers are the only objects that use this.
* Outdated `_1` prefixes are removed from container type bitflags.
* Some unused vars are removed.
* A mostly unrelated change: chemicals logging is now more standardized and uses a process of reagents instead of copypasted code. It got in my way, so it had to die.

### CODE REVIEW

**Comment by praisenarsie** [comment]
> Reagent dispensers (water tank, etc) no longer rely on snowflake checks, it's now possible to make any container have dispenser-like behavior.

hot

File: code/modules/reagents/chemistry/holder.dm [hunk 1]
```diff
@@ -60,6 +60,18 @@
 	if(my_atom && my_atom.reagents == src)
 		my_atom.reagents = null
 
+
+// Used in attack logs for reagents in pills and such
+/datum/reagents/proc/log_list()
+	var/data = list()
+	if(reagent_list && reagent_list.len) //find a reagent list if there is and check if it has entries
```
Comment (Cyberboss) [inline code/modules/reagents/chemistry/holder.dm]: move the declaration of data inside this if

File: code/modules/reagents/reagent_containers/glass.dm [hunk 2]
```diff
@@ -56,31 +56,29 @@
...
-			to_chat(user, "<span class='notice'>[target] is full.</span>")
+		if(reagents.holder_full())
+			to_chat(user, "<span class='warning'>[src] is full.</span>")
 			return
 
-
-		var/trans = reagents.trans_to(target, amount_per_transfer_from_this)
-		to_chat(user, "<span class='notice'>You transfer [trans] unit\s of the solution to [target].</span>")
+		var/trans = target.reagents.trans_to(src, amount_per_transfer_from_this)
+		to_chat(user, "<span class='notice'>You fill [src] with [trans] unit\s of the contents of [target].</span>")
```
Comment (Cyberboss) [inline code/modules/reagents/reagent_containers/glass.dm]: Does the \s macro work with this?

**Review by Cyberboss — DISMISSED** [review]

**Comment by Cyberboss** [comment]
Make a changelog, also you neglected to mention that microwave_act now works on anything with reagents

File: code/modules/reagents/reagent_containers/glass.dm (hunk 2 above)
Comment (ACCount12) [reply code/modules/reagents/reagent_containers/glass.dm]: Yep, it checks the last [] value as far as I know.

**Comment by ACCount12** [comment]
I don't see anything changelog worthy, it's all internal changes unless something breaks.

The microwave_act - yep, I forgot to mention that one. It used the check for no obvious reason, and I tried to get rid of unnecessary `is_open_container` checks.

**Comment by Tlaltecuhtli** [comment]
this 90% broke slime extracts delete pls

**Comment by ACCount12** [comment]
Ah. This is because I fixed being able to extract reagents out of food with syringes. Silme extracts are on the same system. This is literally one line fix, @Tlaltecuhtli use grinder in the meantime.

**Comment by ACCount12** [comment]
Should be all set now. Any more bugs?

**Comment by arsserpentarium** [comment]
is integrated injector still able to drain reagents? If not, please, make integrated grinder!

**Comment by ACCount12** [comment]
Conflicts resolved, and there are no more bugs surfaced from the test merge. Should be ready.

File: code/game/atoms.dm [hunk 3]
```diff
@@ -194,27 +194,28 @@
...
-	return container_type & INJECTABLE_1
+	return reagents ? (container_type & (INJECTABLE | REFILLABLE)) : FALSE
 
 /atom/proc/is_drawable(allowmobs = TRUE)
-	if(is_injectable(allowmobs)) //Everything that can be injected can also be drawn from, but not vice versa
-		return TRUE
-	return container_type & DRAWABLE_1
+	if(allowmobs && isliving(src))
+		var/mob/living/L = src
+		return L.can_inject()
```
Comment (duncathan) [inline code/game/atoms.dm]: shouldn't this behavior be on `/mob/living/is_drawable()`?

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 4]
```diff
@@ -90,6 +75,24 @@
 			bro.cell.use(30)
 			addtimer(CALLBACK(reagents, /datum/reagents.proc/add_reagent, refill, trans), 600)
 
+	else if(target.is_drainable()) //A dispenser. Transfer FROM it TO us.
+		if (!is_refillable())
+			to_chat(user, "<span class='warning'>[target]'s tab isn't open!</span>")
```
Comment (duncathan) [inline code/modules/food_and_drinks/drinks/drinks.dm]: shouldn't this be `src` instead of `target`?

**Review by duncathan — CHANGES_REQUESTED** [review]
simple things

**Comment by ACCount12** [comment]
@duncathan fixed.

**Pull request merged**